test/cql-pytest/test_compaction_tombstone_gc.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+from util import new_materialized_view, new_test_table
+import nodetool
+import time
+
+# sleep to let a ttl (of `seconds`) expire and
+# the commitlog minimum gc time, in seconds,
+# to be greater than the tombstone deletion time
+def sleep_till_whole_second(seconds=1):
+    t = time.time()
+    time.sleep(seconds - (t - int(t)))
+
+def test_tombstone_gc_with_conflict_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Regression test for fixed https://github.com/scylladb/scylladb/issues/20423
+    """
+    schema = "k int, v int, primary key (k, v)"
+    with new_test_table(cql, test_keyspace, schema, extra="with gc_grace_seconds = 0") as table:
+        with nodetool.no_autocompaction_context(cql, table):
+            # Insert initial data into the base table:
+            # Row 1 is expected to be garbage collected after expiration
+            # This test case tests the ttl case, while `test_tombstone_gc_with_delete_in_memtable`
+            # test also explicit deletion.
+            cql.execute(f"insert into {table} (k, v) values (1, 1) using timestamp 1 and ttl 1")
+            cql.execute(f"insert into {table} (k, v) values (1, 2) using timestamp 2")
+            nodetool.flush(cql, table)
+            # Delete row 2, this deletion is expected to be kept
+            # when we insert backdated live data into the memtable
+            cql.execute(f"delete from {table} using timestamp 3 where k=1 and v=2")
+            # Flush all tables explicitly now, since this is skipped in the test on purpose by next major compaction.
+            nodetool.flush_all(cql)
+
+            sleep_till_whole_second()
```

Comment:
This seems to me, indeed, the correct amount of sleep for the TTL (and also, sadly, the not-really-behaving-likezero gc_grace_seconds), but earlier you had here sleep(2)... Was it unnecessary and sleep(1) was enough (and, as you did, even less than 1)? I won't complain :-)

Reply:
I was too conservative with sleep(2) before, and I didn't try to optimize it...
sleep till the next whole second is needed since gc_clock resolution is 1 second
```c++
class gc_clock final {
public:
    using base = seastar::lowres_system_clock;
    using rep = int64_t;
    using period = std::ratio<1, 1>; // seconds
    using duration = std::chrono::duration<rep, period>;
    using time_point = std::chrono::time_point<gc_clock, duration>;
```